oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/ElasticResultRowAsyncIterator.java

```diff
@@ -275,7 +279,9 @@ class ElasticQueryScanner {
                                 .highlight(highlight)
                                 // use a smaller size when the query contains aggregations. This improves performance
                                 // when the client is only interested in insecure facets
-                                .size(needsAggregations.get() ? Math.min(SMALL_RESULT_SET_SIZE, getFetchSize(requests)) : getFetchSize(requests));
+                                .size(needsAggregations.get() ? Math.min(SMALL_RESULT_SET_SIZE, getFetchSize(requests)) : getFetchSize(requests))
+                                // consistently route the same queries to the same shard copy (primary or replica) within the shard set
+                                .preference(sessionId);
```

Comment:
modify/add a UT to check for the sessionId?

Reply:
added simple unit test to verify preference is set and does not change across invocation of the same query